src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -144,7 +145,9 @@ def bulk_record(
             indexer_cache.set_many(new_results_to_cache, use_case_id.value)
             return cache_key_results.merge(db_read_key_results)
 
-        with writes_limiter.check_write_limits(use_case_id, db_write_keys) as writes_limiter_state:
+        with writes_limiter.check_write_limits(
+            use_case_id, db_write_keys, _INDEXER_DB
+        ) as writes_limiter_state:
```

Comment:
I think the rate limiter partition should be an explicit field in `MetricsIngestConfiguration`. (thus two fully different config objects one for spanner performance and one for postgres performance).
The reason is that by using the DB backend you are creating an implicit assumption the correctness of your system depends on: there is one rate limiter per DB backend.
While this is true, it is not intuitive. The rate limiter may be disconnected from the db backend. It owuld be hard to remember about this issue. We messed up the cache key exactly because of a similar issue.
If instead you had two `MetricsIngestConfiguration` and each had to define the rate limiter partition, when writing the config you will be required to think about this problem, thus making it less likely to make mistakes.